src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java

```diff
@@ -263,14 +258,44 @@ private void sendEmail(@NonNull Person recipient, @NonNull List<Notification> no
             .distinct()
             .collect(Collectors.toList());
 
+    Map<String, String> roleForNegotiation = new HashMap<>();
+
+    // Iterate over notifications to populate the map
+    for (Notification notification : notifications) {
+      String negotiationId = notification.getNegotiation().getId();
+      String role = extractRoleFromNotificationMessage(notification);
+      roleForNegotiation.put(negotiationId, role);
+    }
+
     context.setVariable("negotiations", negotiations);
     context.setVariable("frontendurl", frontendurl);
+    context.setVariable("roleForNegotiation", roleForNegotiation);
 
     String emailContent = templateEngine.process("email-notification", context);
 
     emailService.sendEmail(recipient, "New Notifications", emailContent);
   }
 
+  private String extractRoleFromNotificationMessage(Notification notification) {
+    String message = notification.getMessage();
+    if (message.matches("New Negotiation .* was added for review\\.")) {
+      return "ROLE_ADMIN";
+    } else if (message.matches("Negotiation .* had a change of status of .* to .*")) {
+      // TODO if status changed to "ACCESS_CONDITIONS_MET" role should be "ROLE_REPRESENTATIVE"
+      // (once notification also goes to REPRESENTATIVE)
+      return "ROLE_RESEARCHER";
+    } else if (message.matches("Negotiation .* had a new post by .*")) {
+      String[] parts = message.split("new post by");
+      String negotiationCreator = notification.getNegotiation().getCreatedBy().getName();
+      String postCreator = parts[1].trim();
+      return (negotiationCreator.equals(postCreator)) ? "ROLE_REPRESENTATIVE" : "ROLE_RESEARCHER";
+    } else if (message.matches("New Negotiation .*")) {
+      return "ROLE_REPRESENTATIVE";
+    } else {
+      return "ROLE_RESEARCHER"; // Default to ROLE_RESEARCHER
+    }
+  }
+
```

Comment:
maybe use switch statement? And feel free to remove the comments as the logic should be clear from the code.